src/gtfs/read.rs

```diff
@@ -445,24 +494,212 @@ fn get_lines_from_gtfs(gtfs_routes: &[Route], read_mode: &RouteReadType) -> Vec<
             lines.push(l);
         },
         RouteReadType::RouteAsNtmRoute => {
-            // TODO Build lines from GTFS routes as routes
-            unimplemented!();
+            let mut routes: HashSet<&Route> = HashSet::new();
+            let mut backward_routes: HashSet<&Route> = HashSet::new();
+            let mut iter = gtfs_routes.iter();
+            while let Some(r1) = iter.next() {
+                if !backward_routes.contains(r1) {
+                    routes.insert(r1);
+                }
+                for r2 in iter.clone() {
+                    if r1.is_same_line(r2) {
+                        backward_routes.insert(r2);
+                        if r1.id > r2.id {
+                            routes.remove(r1);
+                            routes.insert(r2);
+                        }
+                    } else {
+                        routes.insert(r2);
+                    }
+                }
+            }
```

Comment:
I don't understand this code.

Reply:
suppose you have 3 routes r2, r21 r3 (r1, r2 are the same line)
we'll have 2 lines (r1, r3) (r1 = the smallest id between r1 and r2)

1. I insert r2 
2. r1<r2 => I replace r2 by r1
3. I insert r3
. finally I have my 2 lines r1, r3

poor algo but if you have better..